jupyterhub/app.py

```diff
@@ -1625,10 +1625,7 @@ def stop(self):
         if not self.io_loop:
             return
         if self.http_server:
-            if self.io_loop._running:
-                self.io_loop.add_callback(self.http_server.stop)
-            else:
-                self.http_server.stop()
+            self.http_server.stop()
```

Comment:
The running check is no longer necessary. It was added when tests ran in a thread, so stop had to be passed via callback. This is no longer the case.